care/users/api/viewsets/users.py

```diff
@@ -286,3 +286,23 @@ def pnconfig(self, request, *args, **kwargs):
                 setattr(user, field, request.data[field])
         user.save()
         return Response(status=status.HTTP_200_OK)
+
+class CheckUsernameSerializer(Serializer):
+    username = CharField()
+
+class CheckUsernameAPIView(GenericAPIView):
+    """
+    Checks availability of username, returns 200 if available, and 400 otherwise.
+    """
+
+    permission_classes = (AllowAny, )
+    serializer_class = CheckUsernameSerializer
+
+    def get(self, request):
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        username = serializer.validated_data['username']
```

Comment:
we could use query parameters here.
get method don't support body according to specs.